lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -159,14 +161,15 @@ def run(verify_result: true, allow_protected_ns: false, prune: true)
         @logger.summary.add_paragraph(ColorizedString.new(warning).yellow)
         success = true
       end
+      [success, error]
     rescue FatalDeploymentError => error
       @logger.summary.add_action(error.message)
       success = false
+      [success, error]
     ensure
       @logger.print_summary(success)
       status = success ? "success" : "failed"
       ::StatsD.measure('all_resources.duration', StatsD.duration(start), tags: statsd_tags << "status:#{status}")
-      success
```

Comment:
ensure blocks only have return values if you explicitly use the `return` keyword